ctapipe/tools/make_irf.py

```diff
@@ -0,0 +1,342 @@
+    create_rad_max_hdu,
+)
+from pyirf.irf import (
+    background_2d,
+    effective_area_per_energy,
+    energy_dispersion,
+    psf_table,
+)
+from pyirf.sensitivity import calculate_sensitivity, estimate_background
+from pyirf.simulations import SimulatedEventsInfo
+from pyirf.spectral import (
+    CRAB_HEGRA,
+    IRFDOC_ELECTRON_SPECTRUM,
+    IRFDOC_PROTON_SPECTRUM,
+    PowerLaw,
+    calculate_event_weights,
+)
+from pyirf.utils import calculate_source_fov_offset, calculate_theta
+
+from ..core import Provenance
+from ..io import TableLoader
+from ..irf import DataBinning, IrfToolBase
+
+PYIRF_SPECTRA = {
+    "CRAB_HEGRA": CRAB_HEGRA,
+    "IRFDOC_ELECTRON_SPECTRUM": IRFDOC_ELECTRON_SPECTRUM,
+    "IRFDOC_PROTON_SPECTRUM": IRFDOC_PROTON_SPECTRUM,
+}
+
+
+class IrfTool(IrfToolBase, DataBinning):
+    name = "ctapipe-make-irfs"
+    description = "Tool to create IRF files in GAD format"
+
+    def make_derived_columns(self, events, spectrum, target_spectrum):
+        events["pointing_az"] = 0 * u.deg
+        events["pointing_alt"] = 70 * u.deg
+
+        events["theta"] = calculate_theta(
+            events,
+            assumed_source_az=events["true_az"],
+            assumed_source_alt=events["true_alt"],
+        )
+
+        events["true_source_fov_offset"] = calculate_source_fov_offset(
+            events, prefix="true"
+        )
+        events["reco_source_fov_offset"] = calculate_source_fov_offset(
+            events, prefix="reco"
+        )
+        events["weights"] = calculate_event_weights(
+            events["true_energy"],
+            target_spectrum=target_spectrum,
+            simulated_spectrum=spectrum,
+        )
+
+        return events
+
+    def get_sim_info_and_spectrum(self, loader):
+        sim = loader.read_simulation_configuration()
+
+        sim_info = SimulatedEventsInfo(
+            n_showers=sum(sim["n_showers"] * sim["shower_reuse"]),
+            energy_min=sim["energy_range_min"].quantity[0],
+            energy_max=sim["energy_range_max"].quantity[0],
+            max_impact=sim["max_scatter_range"].quantity[0],
+            spectral_index=sim["spectral_index"][0],
+            viewcone=sim["max_viewcone_radius"].quantity[0],
+        )
+
+        return sim_info, PowerLaw.from_simulation(
+            sim_info, obstime=self.obs_time * u.Unit(self.obs_time_unit)
+        )
+
+    def setup(self):
+        opts = dict(load_dl2=True, load_simulated=True, load_dl1_parameters=False)
+        reduced_events = dict()
+        for kind, file, target_spectrum in [
+            ("gamma", self.gamma_file, PYIRF_SPECTRA[self.gamma_sim_spectrum]),
+            ("proton", self.proton_file, PYIRF_SPECTRA[self.proton_sim_spectrum]),
+            ("electron", self.electron_file, PYIRF_SPECTRA[self.electron_sim_spectrum]),
+        ]:
+            with TableLoader(file, **opts) as load:
+                Provenance().add_input_file(file)
+                table = self._make_empty_table()
```

Comment:
The purpose of `_make_empty_table` is to specify the expectation of what columns need to be present for downstream code, so that it is easier to add readers from other formats later and not depend on the structure of a particular test file.